src/backend/vulkan/src/lib.rs

```diff
@@ -650,11 +652,65 @@ impl queue::QueueFamily for QueueFamily {
     }
 }
 
+pub struct PhysicalDeviceMultiviewProperties {
+    pub max_multiview_view_count: u32,
+    pub max_multiview_instance_index: u32,
+}
+
+impl From<vk::PhysicalDeviceMultiviewProperties> for PhysicalDeviceMultiviewProperties {
+    fn from(other: vk::PhysicalDeviceMultiviewProperties) -> Self {
+        PhysicalDeviceMultiviewProperties {
+            max_multiview_view_count: other.max_multiview_view_count,
+            max_multiview_instance_index: other.max_multiview_instance_index,
+        }
+    }
+}
+
+pub struct PhysicalDeviceProperties {
+    properties: vk::PhysicalDeviceProperties,
+    multiview: Option<PhysicalDeviceMultiviewProperties>,
+}
+
+impl PhysicalDeviceProperties {
+    pub fn load(instance: &Instance, device: &vk::PhysicalDevice, extensions: &[vk::ExtensionProperties]) -> Self {
+        fn supports_extension(extensions: &[vk::ExtensionProperties], extension: &CStr) -> bool {
+            extensions.iter()
+                .any(|ep| unsafe { CStr::from_ptr(ep.extension_name.as_ptr()) } == extension)
+        }
+
+        if let Some(ref get_device_properties) = instance.raw.get_physical_device_properties {
+            let mut properties2_builder = vk::PhysicalDeviceProperties2KHR::builder()
+                .properties(vk::PhysicalDeviceProperties::builder().build());
+            let mut multiview = None;
+
+            // Add extension infos to the p_next chain
+            if supports_extension(extensions, *KHR_MULTIVIEW) {
```

Comment:
something is not right here. Why are we checking for this extension in two places?

Reply:
This check is performed only once during the initialization of `PhysicalDevice`s, I have now replaced the same check within `PhysicalDevice::features()`: https://github.com/gfx-rs/gfx/pull/3323/files#diff-44198a1b71ecb41e695db8b6967cd689R1028